sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1062,8 +1069,10 @@ public Instant getWatermark() {
         return initialWatermark;
       }
 
+      //watermark increases as current_timestamp, in case of latency,
+      //developers need to either provide a customized watermarkFn, or tune Triggers.
       return source.spec.getWatermarkFn() != null
-          ? source.spec.getWatermarkFn().apply(curRecord) : curTimestamp;
```

Comment:
Often the consumers of the streams may not be aware of these timestamps or might just want KafkaIO to do the most sensible thing. I am not sure having a default implementation that marks most records as late and expecting user to handle the correctly is a great policy.

To summarize, my proposal (reasons are discussed earlier):
  - Default to processing timestamps. i.e. KafkaRecord will have the timestamp provided by the Kafka (if any) and 'curTimestamp' and 'watermark' don't change from previous behavior. The simplest option.
  - `OR` Handle LogAppendTimestamp properly to return appropriate watermark (essentially `min(log_append_time for each partition read)`. We can do this in this PR or in another one.

I think this is an important aspect of this PR. Would love to hear yours and others thoughts too. cc @kennknowles @dhalperi @davorbonaci